tinygrad/ops.py

```diff
@@ -337,9 +337,13 @@ class UOps(HashEnum):
 @dataclass(frozen=True, eq=False)
 class UOp(MathTrait):
   op: UOps
-  dtype: Optional[DType] = None
+  dtype: DType = dtypes.void
   src: Tuple[UOp, ...] = tuple()
   arg: Any = None
+  @classmethod
+  def unwrap_dtype(cls, dtype:Optional[DType]) -> Any:
```

Comment:
I need this to let NOp still (safely) accept Optional[DType].
We really need a better abstraction for UOp <-> NOp, I don't think pure inheritance in the answer.

Reply:
https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype

Reply:
https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype